docs/guides/single-fetch.md

```diff
@@ -0,0 +1,155 @@
+---
+title: Single Fetch
+---
+
+# Single Fetch
+
+Remix introduced support for "Single Fetch" ([RFC][rfc]) behind the [`future.unstable_singleFetch`][future-flags] flag in [`v2.9.0`][2.9.0] which allows you to opt-into this behavior which will become the default in Remix v3.
+
+## Overview
+
+When you enable Single Fetch, Remix will make a single HTTP call to your server on client-side transitions, instead of multiple HTTP calls in parallel (one per loader). If you are currently returning `Response` instances from your loaders (i.e., `json`/`defer`) then you shouldn't _need_ to make any changes to your app code, but please read through the "breaking" changes below to be aware of some of the underlying behavior changes.
+
+### Breaking Changes
+
+- Single fetch uses a new streaming format under the hood via [`turbo-stream`][turbo-stream], which means that we can stream down more complex data than just JSON, including `Date`, `Map`, `Set`, `Promise`, `RegExp`, and `BigInt` instances
```

Comment:
If I remember correctly, `Error` is also included in turbo-stream, even custom errors as far as you define them globally on the browser.

Reply:
Not all, but a small subset that has the generic new Error(message) constructor signature + being available globally in both environments.

Reply:
Yeah that's a good call out to add to the docs - we serialize `error.name` if it exists, and then if `global[serializedName]` exists on the client we'll re-construct with that instead of `new Error`

Reply:
Added a note in [6f55ec4](https://github.com/remix-run/remix/pull/9093/commits/6f55ec4dd6797493a425aab6225ff001b379db37)